### ISSUE TITLE
Optionally exit cell/row edit mode

### ISSUE THREAD

**Issue by ddincheva** [IgniteUI/igniteui-angular#4135]
## Is your feature request related to a problem? Please describe.
The user should be able to perform validation over the edited cell/row and to escape or keep cell/row in edit mode. #4096

## Describe the solution you'd like
Should provide a way for the user to determine when the cell to stays in edit mode and when to escape, based on the edited value. Also it is good to consider the same behavior and for row editing. 
Currently we expose a number of events: onCellEditEnter, onCellEdit, onCellEditCancel; onRowEditEnter, onRowEdit, onRowEditCancel; which are cancel-able, so may be it is good to reconsider how this events interact with each other and to define the grid state after each performed action.

**Comment by damyanpetev** [comment]
@ddincheva As far as I understand from the reply and sample on the linked issue was related to `onCellEdit` being cancellable but not keeping the cell in edit mode - merely preventing the commit?
That seems to not be the case for a while now, looking at the sample under https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/cell_editing.html#cell-validation-on-edit-event. Same applies for keeping row in edit mode cancelling its respective event.

Additional improvements are also made in https://github.com/IgniteUI/igniteui-angular/pull/7986 such as properly keeping row in edit mode when a cell is cancelled.

If I got the description right, maybe this issue is already resolved or should be closed by the PR mentioned above?

**Comment by ddincheva** [comment]
@damyanpetev the described problem below focuses more on the fact if you try to submit a cell:
- by pressing ENTER and its value is not valid so you cancel the cellEdit event - the cell still stays in edit mode and its value is not submitted;
- by pressing TAB and its value is not valid so you cancel the cellEdit event - the cell exit edit mode and ist value is not submitted and the next editable cell enters edit mode;
- by Click on another cell and its value is not valid so you cancel the cellEdit event -  the cell exit edit mode and ist value is not submitted and if the current cell that is clicked/if is editable/ enters edit mode.
So to sum up the issue is related to the fact that we have different behaviors when performing a cell editing and try to validate its value. This problem is also reproducible when doing a row editing.
